src/compiler/crystal/semantic/ast.cr

```diff
@@ -57,9 +57,9 @@ module Crystal
 
   # Fictitious node to represent a tuple indexer
   class TupleIndexer < Primitive
-    getter index : Int32
+    getter index : Int32 | Range(Int32, Int32)
```

Comment:
Maybe use `alias Index = Int32 | Range(Int32, Int32)` so that union isn't repeated eveywhere?

Reply:
Well, "everywhere": it's just in a few places. But still...